include/simdjson/parsedjsoniterator.h

```diff
@@ -92,8 +99,14 @@ class ParsedJson::BasicIterator {
 
   inline bool is_integer() const { return get_type() == 'l'; }
 
+  inline bool is_unsigned_integer() const { return get_type() == 'u'; }
```

Comment:
should it be any positive integer?

In that case, I think I could easily add it to the fuzzer.

Reply:
Hmm, your suggestion is like below?

```cpp
inline bool is_unsigned_integer() const {
  return get_type() == 'u' || (is_integer() && get_integer() >= 0);
}
```

This behavior seems consistent. But there is a little performance tradeoff.

Even if this `is_unsigned_integer()` returns true, only safe way to read the value is `get_unsigned_integer()` because `get_integer()` happens overflow for a large unsigned integer.
In such a situation, I think it's enough to check if the element type is 'u', so `(is_integer() && get_integer() >= 0)` is redundant.

Even so, should I do it?

Reply:
@saka1 I don't think that @ioioioio meant that you had to change something in the code/logic. However, what about documenting what is expected? That is, if I have the value "2"... what are the returned values for is_integer and is_unsigned_integer? (i.e., what should the users of the API expect?)

Reply:
> what about documenting what is expected?

I got it. I added some comments  https://github.com/lemire/simdjson/pull/295/commits/e7cb3d6ec2dc835b71f9e66c47381a5267ae8b6d.
@lemire @ioioioio How is it?

Reply:
well, in fact I was asking if we wanted a behaviour similar to [rapidjson's](http://rapidjson.org/md_doc_tutorial.html#QueryNumber). Yes, something like what you described. Seems like it does not matter. In any case, Daniel did point out a real issue. The correction looks good to me.

I also thought that a behaviour like rapidjson's would be easier to test with the [fuzzer](https://github.com/ioioioio/fuzzyjson). On a second thought, it should not be a lot more complicated.